mlflow/models/utils.py

```diff
@@ -142,28 +142,20 @@ def _handle_dataframe_nans(df: pd.DataFrame):
 
         def _coerce_to_pandas_df(input_ex):
             if isinstance(input_ex, dict):
-                if all(_is_scalar(x) for x in input_ex.values()):
-                    input_ex = pd.DataFrame([input_ex])
-                elif all(isinstance(x, (str, list)) for x in input_ex.values()):
-                    for value in input_ex.values():
-                        if isinstance(value, list) and not all(_is_scalar(x) for x in value):
-                            raise TypeError(
-                                "List values within dictionaries must be of scalar type."
-                            )
-                    input_ex = pd.DataFrame(input_ex)
-                else:
-                    raise TypeError(
-                        "Data in the dictionary must be scalar or of type numpy.ndarray"
+                # We need to be compatible with infer_schema's behavior, where
+                # it infers each value's type directly.
+                if all(
+                    isinstance(x, str) or (isinstance(x, list) and all(_is_scalar(y) for y in x))
+                    for x in input_ex.values()
+                ):
+                    _logger.info(
+                        "We convert input dictionaries to pandas DataFrames by assuming "
+                        "each key is a column, and in total, one row of data. If you "
```

Comment:
`in total, one row of data` sounds a bit confusing.

Reply:
/ai Could you rewrite the sentence to make it clear?

Reply:
Do we expect a dict like this `{"a": 0, "b": 1}`?

Reply:
Yes, it's acceptable

Reply:
The only difference here is that for data like {"a": "a", "b": ["a", "b", "c"]} now is converted to one row instead of three rows. And we support other value types that are not supported before.

Reply:
Got it. Adding an example might help clarify what the comment means.